haskell/test/Test/Ligo/BaseDAO/Proposal/Propose.hs

```diff
@@ -521,3 +522,44 @@ proposalBoundedValue originateFn = do
     call dodDao (Call @"Propose") params
       & expectCustomErrorNoArg #mAX_PROPOSALS_REACHED
 
+proposalStressTest
+  :: forall caps base m.
+    ( MonadNettest caps base m
+    , HasCallStack
+    )
+  => (ConfigDesc Config -> OriginateFn m) -> m ()
+proposalStressTest originateFn = do
+  DaoOriginateData {..} <- originateFn
+    ((ConfigDesc $ configConsts { cmMaxProposals = Just 1000 }) >>-
+     (ConfigDesc $ Period 100) >>- testConfig) defaultQuorumThreshold
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Freeze") (#amount .! 100000)
+
+  originLevel <- getOriginationLevel dodDao
+
+  -- Ensure that we are in a proposal period, or wait, if we are not yet there.
+  let ensureProposalPeriod = do
+        nowLevel <- getLevel
+        let periodNum = div ((nowLevel + 10) - originLevel) dodPeriod
+        when (even periodNum) $ do
+          let waitTill = originLevel + (periodNum + 1)*dodPeriod
+          advanceToLevel waitTill
+
+  -- Create a thousand proposals in batchs of 100
+  forM_ [0..(9 :: Natural)] $ \count -> do
+    let multiplier = 100
+    let range = [( count * multiplier + 1 ) .. ((count+1) * multiplier )]
+    ensureProposalPeriod
+
+    withSender dodOwner1 $ inBatch $ do
+      traverse_ (\c ->
+        let params = ProposeParams
+              { ppFrozenToken = 10
+              , ppProposalMetadata = lPackValueRaw c
+              , ppFrom = dodOwner1
+              }
+        in call dodDao (Call @"Propose") params) range
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Freeze") (#amount .! 1)
```

Comment:
Should we try to vote/drop_proposal on some of these proposals to make sure it is fine?

Reply:
I think it does not make much of a difference since the storage needs to be decoded no matter which entrypoint we call.

Reply:
I see, since the concern is about the storage decoding, I'm fine with this.